server/ocsp.go

```diff
@@ -400,11 +417,52 @@ func (srv *Server) NewOCSPMonitor(tc *tls.Config) (*tls.Config, *OCSPMonitor, er
 			}, nil
 		}
 
-		// GetClientCertificate returns a certificate that's presented to a
-		// server.
-		tc.GetClientCertificate = func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return &cert, nil
+		// Check whether need to verify staples from a client connection depending on the type.
+		switch kind {
+		case typeStringMap[ROUTER], typeStringMap[GATEWAY], typeStringMap[LEAF]:
+			tc.VerifyConnection = func(s tls.ConnectionState) error {
+				oresp := s.OCSPResponse
+				if oresp == nil {
+					return fmt.Errorf("missing OCSP Staple")
```

Comment:
All of these error messages should surely identify which staple is missing, making clear that it's to do with staples from a peer, rather than the local cert itself.

Reply:
We can infer some of this from the debug lines from the server, for example here is how it would like when a peer has the bad OCSP staple.

```
[91176] [DBG] 127.0.0.1:52753 - rid:3 - Starting TLS route client handshake
[91176] [ERR] 127.0.0.1:52753 - rid:3 - TLS route handshake error: bad status for OCSP Staple
[91176] [INF] 127.0.0.1:52753 - rid:3 - Router connection closed: TLS Handshake Failure
```
But debug is silent by default so we can do maybe this

```
fmt.Errorf("%s client missing OCSP Staple", kind)
```

So that it reads

```
fmt.Errorf("route client missing OCSP Staple", kind)
```
```

Reply:
👍